contracts/ecash/src/storage.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use cw_storage_plus::{Index, IndexList, IndexedMap, UniqueIndex};
+use nym_ecash_contract_common::{
+    blacklist::{BlacklistProposal, BlacklistedAccount},
+    spend_credential::EcashSpentCredential,
+};
+
+// storage prefixes
+const SPEND_CREDENTIAL_PK_NAMESPACE: &str = "ecsc";
+const SPEND_CREDENTIAL_BLINDED_SERIAL_NO_IDX_NAMESPACE: &str = "ecscn";
+
+// paged retrieval limits for all queries and transactions
+pub(crate) const SPEND_CREDENTIAL_PAGE_MAX_LIMIT: u32 = 75;
+pub(crate) const SPEND_CREDENTIAL_PAGE_DEFAULT_LIMIT: u32 = 50;
+
+pub(crate) struct SpendCredentialIndex<'a> {
+    pub(crate) blinded_serial_number: UniqueIndex<'a, String, EcashSpentCredential>,
+}
+
+// IndexList is just boilerplate code for fetching a struct's indexes
+// note that from my understanding this will be converted into a macro at some point in the future
+impl<'a> IndexList<EcashSpentCredential> for SpendCredentialIndex<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<EcashSpentCredential>> + '_> {
+        let v: Vec<&dyn Index<EcashSpentCredential>> = vec![&self.blinded_serial_number];
+        Box::new(v.into_iter())
+    }
+}
+
+// spent_credentials() is the storage access function.
+pub(crate) const fn spent_credentials<'a>(
```

Comment:
either use a simple `Map`, because I reckon we don't need the extra index, or the raw bytes i described before

Reply:
Fixed in 3fe111c20036562f9dd8cc819c57234a914993ad